conversation/generate_chat_logs.py3

```diff
@@ -0,0 +1,132 @@
+#!/usr/local/bin/python3
+#
+# Given a CSV file containing utterances to simulate a user's questions, and a set of
+# input parameters, this script will call the Watson Conversation /message API and 
+# simulate an interaction between a user and the Conversation service. The result will
+# include data in the Improve tab of the Watson Conversation Service UI.
+#
+# The CSV file format allows you to include a sequence of utterances 
+# in the same conversation. Place each utterance on a separate line.
+# To start a new conversation, you should include a line with the string :init:
+#
+# For example, a CSV to simulate two conversations with the Car Dashboard Demo
+# sample application # could be as simple as these four lines:
+# question,answer_regex,"intents"
+# turn lights on
+# :init:
+# hit the brake
+#
+#
+# If you don't have python3 requests installed, be sure to run pip3 install requests
+
+#!/usr/local/bin/python3
+import sys
+import json
+import time
+import urllib.request
+import urllib.error
+import base64
+import datetime
+import re
+import ssl
+import csv
+import requests
+import argparse
+
+test_result = True
+preProvisionedInstance = False
+
+parser = argparse.ArgumentParser()
+parser.add_argument("endpoint")
+parser.add_argument("service_creds")
+parser.add_argument("workspace_id")
+parser.add_argument("question_csv_file")
+parser.add_argument("-c", "--customer_id")
+parser.add_argument("-d", "--deployment_id")
+parser.add_argument("-u", "--user_id")
+args = parser.parse_args()
+
+endpoint = args.endpoint
+service_creds = args.service_creds
+workspace_id = args.workspace_id
+auth_header = 'Basic ' + base64.b64encode(bytes(service_creds,"utf-8")).decode("ascii")[:-1]
+version = "?version=2017-05-03"
+gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+
+if True:
+    csv_reader = csv.DictReader(open(args.question_csv_file),escapechar="\\")
+    deployTag = {"deployment": args.deployment_id} if args.deployment_id is not None else None
+    userID = {"user_id": args.user_id} if args.user_id is not None else None
+    metadataValue = None
+    if deployTag is not None:
+        if userID is not None:
+            metadataValue = deployTag.copy()
+            metadataValue.update(userID)
+        else:
+            metadataValue = deployTag
+    else:
+        if userID is not None:
+            metadataValue = userID
+    context = {"metadata": metadataValue} if metadataValue is not None else None
+    prevContext = None
+    turnCount = 0
+
+    for turn in csv_reader:
+        turnCount = turnCount + 1
+        if turn["question"][0] == ";":
+            continue
+        if turn["question"] == ":init:":
+            context = {"metadata": metadataValue} if metadataValue is not None else None
+            turnCount = 0
+            continue
+
+        message = { "input": { "text": turn["question"] } ,"alternate_intents": True}
```

Comment:
If you use the SDK you won't have worry about TLS2 or building the REST API on your own.
I imagine something like

```py
conversation = ConversationV1(
    username=os.getenv("CONVERSATION_USERNAME"),
    password=os.getenv("CONVERSATION_PASSWORD"),
    version='2017-04-21',
    default_headers= { 'X-Watson-Metadata' : 'customer_id=' + args.customer_id })


workspace_id = '506e4a2e-3d5d-4dca-b374-38edbb4139ab'
if os.getenv("conversation_workspace_id") is not None:
    workspace_id = os.getenv("conversation_workspace_id")

response = conversation.message(workspace_id=workspace_id, input={
    'text': 'What\'s the weather like?'})
print(json.dumps(response, indent=2))